src/scripts/main.js

```diff
@@ -1,3 +1,310 @@
+
+      case SALARY.toLowerCase():
+        td.innerText = salaryToString(data.get(SALARY.toLowerCase()));
+        break;
+    }
+    newRow.append(td);
+  }
+
+  let counterFree = 0;
+
+  for (const free of newRow.children) {
+    if (free.innerText) {
+      counterFree++;
+    }
+  }
+
+  if (counterFree !== newRow.children.length) {
+    informMessage('warning', messageFullData);
+  } else {
+    document.querySelector('tbody').append(newRow);
+    form.reset();
+    informMessage('success', messageSuccess);
+  }
+});
+
+document.querySelector('tbody').addEventListener('dblclick', e => {
+  const cell = e.target;
+
+  cell.parentElement.className = '';
+
+  const input = document.createElement('input');
+
+  input.className = 'cell-input';
+  input.value = cell.innerText;
+  cell.replaceWith(input);
+  input.focus();
+
+  input.addEventListener('keydown', ev => {
+    if (ev.key === 'Enter') {
+      input.blur();
+    }
+  });
+
+  input.addEventListener('blur', ev => {
+    if (!input.value) {
+      input.replaceWith(cell);
+    } else {
+      cell.innerText = input.value;
+      input.replaceWith(cell);
+    }
+  });
+});
+
+function listOfOffices(table) {
+  const listOffices = {};
+  const header = [...table.firstElementChild.firstElementChild.children];
+
+  const rows = [...table.rows].slice(0, -1);
+
+  for (let i = 0; i < header.length; i++) {
+    if (header[i].textContent !== OFFICE) {
+      continue;
+    }
+
+    for (const row of rows) {
+      if (listOffices[row.children[i].textContent]) {
+        continue;
+      } else {
+        listOffices[row.children[i].textContent] = true;
+      }
+    }
+  }
+
+  return Object.keys(listOffices);
+}
+
+function sortBy(tbEl, thEl, value) {
+  for (let i = 0; i < thEl.length; i++) {
+    if (thEl[i].textContent !== value) {
+      continue;
+    }
+
+    tbEl.sort((a, b) => {
+      switch (value) {
+        case NAME:
+        case POSITION:
+        case OFFICE:
+          return a.children[i].textContent
+            .localeCompare(b.children[i].textContent);
+
+        case AGE:
+          return +a.children[i].textContent - +b.children[i].textContent;
+
+        case SALARY:
+          return salaryToNumber(a.children[i].textContent)
+            - salaryToNumber(b.children[i].textContent);
+      }
+    });
+  }
+
+  return tbEl;
+}
+
+function salaryToNumber(salary) {
+  return +salary.slice(1).split(',').join('');
+}
+
+function salaryToString(str) {
+  let result = '';
+  let counter = 0;
+
+  for (let i = str.length - 1; i >= 0; i--) {
+    result = str[i] + result;
+    counter++;
+
+    if (counter % 3 === 0 && counter !== str.length) {
+      result = ',' + result;
+    }
+  }
+
+  return `$${result}`;
+}
+
+function informMessage(type, text) {
+  const div = document.createElement('div');
+
+  div.className = 'notification';
+  div.classList.add(type);
+  div.dataset.qa = 'notification';
+  div.innerText = text;
+  div.style.color = 'black';
+  form.prepend(div);
```

Comment:
```suggestion
  document.body.append(div);
```
it can make better view for notification
now it is not good
![chrome-capture-2023-9-26](https://github.com/mate-academy/js_employees_table_DOM/assets/33029864/ec55408d-f907-4d8a-8016-d9cbca0e045d)